sound/soc/sof/ipc4-topology.c

```diff
@@ -1976,12 +1978,36 @@ static int sof_ipc4_control_setup(struct snd_sof_dev *sdev, struct snd_sof_contr
 	return 0;
 }
 
+static int sof_ipc4_pipeline_send_kcps(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget,
+				       __s32 adjust)
+{
+	struct sof_ipc4_msg msg = {
+		.primary = SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_MOD_LARGE_CONFIG_SET) |
+			SOF_IPC4_MSG_TARGET(SOF_IPC4_MODULE_MSG) |
+			SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST) |
+			SOF_IPC4_MOD_ID(SOF_IPC4_MOD_INIT_BASEFW_MOD_ID) |
+			SOF_IPC4_MOD_INSTANCE(SOF_IPC4_MOD_INIT_BASEFW_INSTANCE_ID),
+		.extension = SOF_IPC4_MOD_EXT_MSG_PARAM_ID(SOF_IPC4_FW_PARAM_REGISTER_KCPS) |
+			SOF_IPC4_MOD_EXT_MSG_FIRST_BLOCK(1) |
+			SOF_IPC4_MOD_EXT_MSG_LAST_BLOCK(1),
+		.data_size = sizeof(adjust),
+		.data_ptr = &adjust,
+	};
+
+	/*
+	 * The last argument is used in sof_ipc4_tx_msg() just as a flag to dump
+	 * IPC data when verbose debugging is enabled
+	 */
```

Comment:
Well, yes and no, it is used to track the payload size in the mailbox, it is used in the IPC low level code to decide if we need to copy something to there or not.
Can you drop this comment?

Reply:
I think it's the `msg.data_size`, that's used for that in the IPC4 case, not the last parameter to `sof_ipc_tx_message_no_reply()`, but sure, I can drop it

Reply:
I stand corrected - I was looking at an older kernel version. Now it's used indeed!